src/Mollie/WC/Payment/Object.php

```diff
@@ -230,7 +230,7 @@ public function unsetActiveMolliePaymentForOrders( $order_id ) {
         $order             = wc_get_order( $order_id );
         $mollie_payment_id = $order->get_meta( '_mollie_payment_id', true );
 
-        if ( $mollie_payment_id == $this->data->id ) {
+        if (is_object($this->data) && $mollie_payment_id == $this->data->id ) {
```

Comment:
I would also check if `$this->data->id` exist, something like this:
`if (is_object($this->data) && isset($this->data->id) && $mollie_payment_id === $this->data->id ) {`